src/main/java/com/thoughtworks/gauge/scan/CustomClassInitializerScanner.java

```diff
@@ -0,0 +1,39 @@
+package com.thoughtworks.gauge.scan;
+
+import java.util.NoSuchElementException;
+import java.util.Set;
+
+import org.reflections.Reflections;
+
+import com.thoughtworks.gauge.ClassInitializer;
+import com.thoughtworks.gauge.CustomClassInitializer;
+import com.thoughtworks.gauge.registry.ClassInitializerRegistry;
+
+public class CustomClassInitializerScanner implements IScanner {
+    @Override
+    public void scan(Reflections reflections) {
+        scanForInitializer(reflections);
+    }
+
+    private void scanForInitializer(Reflections reflections) {
+        Set<Class<?>> initializers = reflections.getTypesAnnotatedWith(CustomClassInitializer.class);
+
+        if (initializers.size() > 1) {
+            throw new ScanningException("Only 1 class can be annotated with @CustomClassInitializer");
+        }
+
```

Comment:
I think it's better to implement a default `ClassInitializer` which we can return from here if there is no `CustomClassInitializer` implemented by the user. So we won't require any null check in `ClassInstanceManager`. What do you think?